src/coreComponents/codingUtilities/TableFormatter.hpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableFormatter.hpp
+ */
+
+#ifndef GEOS_COMMON_TABLEFORMATTER_HPP
+#define GEOS_COMMON_TABLEFORMATTER_HPP
+
+#include "codingUtilities/TableData.hpp"
+#include "codingUtilities/TableLayout.hpp"
+
+namespace geos
+{
+
+// Class for formatting table data
+class TableFormatter
+{
+public:
+
+  /**
+   * @brief Constructor by default
+   */
+  TableFormatter() = default;
+
+  /**
+   * @brief Construct a new Table Formatter from a tableLayout
+   * @param tableLayout Contain all column names and optionnaly the table title
+   */
+  TableFormatter( TableLayout const & tableLayout );
+
+  /**
+   * @brief Fill the vector (m_column) in tableData with values from m_rows in tableLayout, storing all values in an unsorted order.
+   * @param columns Vector of columns to be filled.
+   * @param tableData Vector of table data.
+   */
+  void fillTableColumnsFromRows( std::vector< TableLayout::Column > & columns,
+                                 std::vector< std::vector< string > > const & tableData ) const;
+
+protected:
+
+  TableLayout m_tableLayout;
+};
+
+class TableCSVFormatter : public TableFormatter
+{
+public:
+  /**
+   * @brief Constructor by default
+   */
+  TableCSVFormatter() = default;
+
+  /**
+   * @brief Construct a new Table Formatter from a tableLayout
+   * @param tableLayout Contain all column names and optionnaly the table title
+   */
+  TableCSVFormatter( TableLayout const & tableLayout );
+
+  /**
+   * @brief Convert the table data to a CSV string.
+   * @param tableData The 1D table data.
+   * @return The CSV string representation of the table data.
+   */
+  string dataToString( TableData const & tableData ) const;
+
+  /**
+   * @return The string with all column names.
+   */
+  string headerToString() const;
+
+  /**
+   * @brief Convert the TableData to a table string.
+   * @param tableData The TableData to convert.
+   * @return The table string representation of the TableData.
+   */
+  string toString( TableData const & tableData ) const;
+
+};
+
+class TableTextFormatter : public TableFormatter
+{
+
+public:
+
+  TableTextFormatter( TableLayout const & tableLayout );
+
+  TableTextFormatter( std::vector< string > const & columnNames );
```

Comment:
Are you using this?

Reply:
Yes the PVT log PR ( tableFunction.cpp) , I've used the these functions

Reply:
Sorry I can't find it. Can you point me where you're using the `TableTextFormatter( std::vector< string > const & columnNames )` constructor?